app/src/test/java/co/nimblehq/compose/crypto/ui/screens/home/HomeScreenTest.kt

```diff
@@ -0,0 +1,283 @@
+        get() = composeAndroidTestRule.activity.getString(R.string.portfolio_card_total_coin_label)
+
+    private val todayProfitLabel: String
+        get() = composeAndroidTestRule.activity.getString(R.string.portfolio_card_today_profit_label)
+
+    private val errorGeneric: String
+        get() = composeAndroidTestRule.activity.getString(R.string.error_generic)
+
+    private val mockGetMyCoinsUseCase = mockk<GetMyCoinsUseCase>()
+    private val mockGetTrendingCoinsUseCase = mockk<GetTrendingCoinsUseCase>()
+
+    private lateinit var viewModel: HomeViewModel
+
+    private var appDestination: AppDestination? = null
+
+    private val totalCoin = "7,273,291"
+    private val todayProfit = "193,280"
+
+    @Before
+    fun setUp() {
+        composeAndroidTestRule.activity.setContent {
+            HomeScreen(
+                viewModel = viewModel,
+                navigator = { destination -> appDestination = destination }
+            )
+        }
+    }
+
+    @Test
+    fun `When enter to HomeScreen, it render the PortfolioCard properly`() {
+        initViewModel()
+
+        with(composeAndroidTestRule) {
+            onNodeWithTag(testTag = homeTitle).assertTextEquals(homeTitle)
+            onNodeWithTag(testTag = totalCoinsLabel).assertTextEquals(totalCoinsLabel)
+            onNodeWithTag(testTag = todayProfitLabel).assertTextEquals(todayProfitLabel)
+            onNodeWithTag(testTag = TestTagCardTotalCoins).assertTextEquals(
+                activity.getString(R.string.coin_currency, totalCoin)
+            )
+            onNodeWithTag(testTag = TestTagCardTodayProfit).assertTextEquals(
+                activity.getString(R.string.coin_currency, todayProfit)
+            )
+        }
+    }
+
+    @Test
+    fun `When enter to HomeScreen and load MyCoins successfully, it render the UI properly`() {
+        every { mockGetMyCoinsUseCase.execute(any()) } returns flowOf(MockUtil.myCoins)
+
+        initViewModel()
+
+        with(composeAndroidTestRule) {
+            onNodeWithTag(testTag = TestTagCoinsLoader).assertIsDisplayed()
+
+            with(MockUtil.myCoins.first()) {
+                onAllNodesWithTag(
+                    testTag = TestTagCoinItemSymbol,
+                    useUnmergedTree = true
+                ).onFirst().assertTextEquals(symbol.uppercase())
+                onAllNodesWithTag(
+                    testTag = TestTagCoinItemCoinName,
+                    useUnmergedTree = true
+                ).onFirst().assertTextEquals(coinName)
+                onAllNodesWithTag(
+                    testTag = TestTagCoinItemPriceChange,
+                    useUnmergedTree = true
+                ).onFirst().onChild()
+                    .assertTextEquals(
+                        activity.getString(
+                            R.string.coin_profit_percent,
+                            priceChangePercentage24hInCurrency.toFormattedString()
+                        )
+                    )
```

Comment:
Can we move this outside by defining a new variable? 😄 

```suggestion
val expectedPriceChange = activity.getString(
                            R.string.coin_profit_percent,
                            priceChangePercentage24hInCurrency.toFormattedString()
                        )
                    .assertTextEquals(expectedPriceChange)
```

Reply:
@hoangnguyen92dn, I'm good at both the current way and defining a new variable, but to make it readable then, we can go with  defining a new variable  🚀